### PR TITLE
feat!(p2p): put networkID as the first sub-string in protocolID

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview

* Changed protocolID representation to `networkID/domain/*sub-domain*/version`, where `sub-domain` is optional.
* Remove dependency on node builder in libs

## Checklist
- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

### CODE REVIEW

File: share/p2p/shrexsub/pubsub.go [hunk 1]
```diff
@@ -15,9 +15,9 @@ import (
 
 var log = logging.Logger("shrex-sub")
 
-// pubsubTopic hardcodes the name of the EDS floodsub topic with the provided suffix.
-func pubsubTopic(suffix string) string {
-	return fmt.Sprintf("/eds-sub/v0.0.1/%s", suffix)
+// PubsubTopic hardcodes the name of the EDS floodsub topic with the provided networkID.
+func PubsubTopicID(networkID string) string {
```
Comment (Wondertan) [inline share/p2p/shrexsub/pubsub.go:19]: Not necessary to make it public, as we there is no scoring for FloodSub

**Review by Wondertan — COMMENTED** [review]
Breaking

We may also use networkID option to check the network type in another PR, as we extended the scope from prefix to networkId

File: share/p2p/shrexsub/pubsub.go (hunk 1 above)
Comment (vgonkivs) [reply share/p2p/shrexsub/pubsub.go]: okay

**Comment by vgonkivs** [comment]
> We may also use networkID option to check the network type in another PR, as we extended the scope from prefix to networkId

Do you mean in chaindID?

File: libs/header/p2p/exchange_test.go [hunk 2]
```diff
@@ -24,10 +24,11 @@ import (
 	headerMock "github.com/celestiaorg/celestia-node/libs/header/mocks"
 	p2p_pb "github.com/celestiaorg/celestia-node/libs/header/p2p/pb"
 	"github.com/celestiaorg/celestia-node/libs/header/test"
-	"github.com/celestiaorg/celestia-node/nodebuilder/p2p"
 )
 
-var privateProtocolID = protocolID(string(p2p.Private))
+const networkID = "private"
+
+var privateProtocolID = protocolID(networkID)
```
Comment (renaynay) [inline libs/header/p2p/exchange_test.go:31]: is this used?

File: share/p2p/shrexeds/options.go [hunk 3]
```diff
@@ -30,9 +30,9 @@ type Parameters struct {
 	// BufferSize defines the size of the buffer used for writing an ODS over the stream.
 	BufferSize uint64
 
-	// protocolSuffix is appended to the protocolID and represents the network the protocol is
+	// networkID is appended to the protocolID and represents the network the protocol is
```
Comment (renaynay) [inline share/p2p/shrexeds/options.go:33]: networkID is prepended *

File: share/p2p/shrexeds/options.go [hunk 4]
```diff
@@ -83,6 +83,6 @@ func WithConcurrencyLimit(concurrencyLimit int) Option {
 	}
 }
 
-func protocolID(protocolSuffix string) protocol.ID {
-	return protocol.ID(fmt.Sprintf("%s%s", protocolPrefix, protocolSuffix))
+func protocolID(networkID string) protocol.ID {
+	return protocol.ID(fmt.Sprintf("%s%s", networkID, protocolPrefix))
```
Comment (renaynay) [inline share/p2p/shrexeds/options.go:87]: let's call `protocolPrefix` just `protocolString` or something and then remove trailing `/`

networkID will also need a leading `/` no?

File: share/p2p/shrexnd/options.go [hunk 5]
```diff
@@ -97,6 +97,6 @@ func WithConcurrencyLimit(concurrencyLimit int) Option {
 	}
 }
 
-func protocolID(protocolSuffix string) protocol.ID {
-	return protocol.ID(fmt.Sprintf("%s%s", protocolPrefix, protocolSuffix))
+func protocolID(networkID string) protocol.ID {
+	return protocol.ID(fmt.Sprintf("%s%s", networkID, protocolPrefix))
```
Comment (renaynay) [inline share/p2p/shrexnd/options.go:101]: same thing w protocolPrefix here ^

**Review by renaynay — CHANGES_REQUESTED** [review]

File: share/p2p/shrexeds/options.go (hunk 4 above)
Comment (vgonkivs) [reply share/p2p/shrexeds/options.go:87]: Why don't you like networkID here?  Prefix/suffix is smth abstract, but here we are explicitly setting networkID

Comment (Wondertan) [reply share/p2p/shrexeds/options.go:87]: Rene is not against networkID and want protocolPrefix to be renamed

Comment (vgonkivs) [reply share/p2p/shrexeds/options.go:87]: oh....sorry
Was responded from phone

Comment (vgonkivs) [reply share/p2p/shrexeds/options.go:87]: > networkID will also need a leading / no?

iirc, no

**Review by Wondertan — APPROVED** [review]

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — DISMISSED** [review]

File: libs/header/p2p/exchange_test.go (hunk 2 above)
Comment (vgonkivs) [reply libs/header/p2p/exchange_test.go]: removed

**Review by Wondertan — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Pull request merged**